src/openforms/registrations/contrib/zgw_apis/plugin.py

```diff
@@ -38,6 +39,46 @@
 logger = logging.getLogger(__name__)
 
 
+def get_variables_properties_from_submission(
+    submission: Submission, connections: dict[str, Any]
+) -> dict[str, Any]:
+    """
+    Extract the values from the submission and map onto the provided properties (eigenshappen).
+    """
+    variables_properties = {}
+
+    # dict of {componentKey: eigenshap} mapping
+    simple_mappings = {conn["component_key"]: conn["eigenshap"] for conn in connections}
+
+    merged_data = dict(
+        submission.submissionvaluevariable_set.filter(
+            key__in=[conn["component_key"] for conn in connections]
+        ).values_list("key", "value")
+    )
+
+    for key, form_value in merged_data.items():
+        if key not in simple_mappings:
+            continue
+
+        eigenshap = simple_mappings.get(key)
+        variables_properties[eigenshap] = form_value
+
+    return variables_properties
+
+
+class MappedVariablePropertySerializer(serializers.Serializer):
+    component_key = serializers.CharField(
+        label=_("Component key"),
+        help_text=_("Key of the Formio.js component to take the value from."),
+    )
```

Comment:
this should (re) use the `openforms.forms.models.form_variable.variable_key_validator` validator.

To make sure that code is located in the right place, this existing validator should be moved into `openforms.formio.validators` too :)